pkg/bot/delete.go

```diff
@@ -100,195 +185,130 @@ func (b *Bot) processDeleteCommand(update tgbotapi.Update, userID int64, r *rada
 	b.sendMessage(editMsg)
 	return false
 }
 
-func (b *Bot) processConfirmationForDelete(update tgbotapi.Update, command *userDeleteMovie) bool {
-	switch update.CallbackQuery.Data {
-	case "DELETEMOVIE_YES":
-		return b.handleDeleteConfirmationYes(update, command)
-	case "DELETEMOVIE_NO_SEARCH":
-		return b.handleDeleteConfirmationNoSearchResults(update, command)
-	case "DELETEMOVIE_NO_LIBRARY":
-		return b.handleDeleteConfirmationNoLibrary(update, command)
-	case "DELETEMOVIE_CANCEL":
-		return b.handleDeleteConfirmationCancel(update, command)
-	default:
-		command.confirmation = false
-		command.movie = nil
-		b.setDeleteMovieState(command.chatID, command)
-		return false
+func (b *Bot) handleDeleteMovieYes(update tgbotapi.Update, command *userDeleteMovie) bool {
+	var movieIDs []int64
+	var deletedMovies []string
+	for _, movie := range command.selectedMovies {
+		movieIDs = append(movieIDs, movie.ID)
+		deletedMovies = append(deletedMovies, movie.Title)
+	}
+	bulkEdit := radarr.BulkEdit{
+		MovieIDs: movieIDs,
 	}
-}
-
-func (b *Bot) handleDeleteConfirmationYes(update tgbotapi.Update, command *userDeleteMovie) bool {
-	command.confirmation = true
-	msg := tgbotapi.NewMessage(command.chatID, "")
 
-	err := b.RadarrServer.DeleteMovie(command.movie.ID, *starr.True(), *starr.False())
+	err := b.RadarrServer.DeleteMovies(&bulkEdit)
 	if err != nil {
-		msg.Text = err.Error()
+		msg := tgbotapi.NewMessage(command.chatID, err.Error())
 		fmt.Println(err)
 		b.sendMessage(msg)
 		return false
 	}
 
+	messageText := fmt.Sprintf("Deleted movies:\n- %v", strings.Join(deletedMovies, "\n- "))
 	editMsg := tgbotapi.NewEditMessageText(
 		command.chatID,
 		command.messageID,
-		fmt.Sprintf("Movie '%v' deleted\n", command.movie.Title),
+		messageText,
 	)
 
 	b.clearState(update)
 	b.sendMessage(editMsg)
 	return true
 }
 
-func (b *Bot) handleDeleteConfirmationNoSearchResults(update tgbotapi.Update, command *userDeleteMovie) bool {
-	searchResults := command.searchResultsInLibrary
-	command.confirmation = false
-	command.movie = nil
-	movies := make([]*radarr.Movie, 0, len(searchResults))
-	for _, value := range searchResults {
-		movies = append(movies, value)
-	}
+func (b *Bot) handleLDeleteMovieSelection(update tgbotapi.Update, command *userDeleteMovie) bool {
+	movieIDStr := strings.TrimPrefix(update.CallbackQuery.Data, "TMDBID_")
+	movie := command.library[movieIDStr]
 
-	sort.SliceStable(movies, func(i, j int) bool {
-		return utils.IgnoreArticles(strings.ToLower(movies[i].Title)) < utils.IgnoreArticles(strings.ToLower(movies[j].Title))
-	})
+	// Check if the movie is already selected
+	if isSelectedMovie(command.selectedMovies, movie.ID) {
+		// If selected, remove the movie from selectedMovies (deselect)
+		command.selectedMovies = removeMovie(command.selectedMovies, movie.ID)
+	} else {
+		// If not selected, add the movie to selectedMovies (select)
+		command.selectedMovies = append(command.selectedMovies, movie)
+	}
```

Comment:
```suggestion
	// If not selected, add the movie to selectedMovies (select)
	command.selectedMovies = append(command.selectedMovies, movie)
	// If selected, remove the movie from selectedMovies (deselect)
	if isSelectedMovie(command.selectedMovies, movie.ID) {
		// If selected, remove the movie from selectedMovies (deselect)
		command.selectedMovies = removeMovie(command.selectedMovies, movie.ID)
	}
```

According to golang best practices we try to avoid else operators.

Reply:
Ah, bullshit. This is not working. But maybe with a helper variable